scripts/configmap/create_cm.sh

```diff
@@ -1,12 +1,164 @@
-k delete cm powerscale-driver-v2.2.0,driver-sidecars,authorization-v2.2.0,authorization-common -n dell-csm-operator
-kubectl create cm powerscale-driver-v2.2.0 --from-file=powerscale/v2.2.0/ -n dell-csm-operator
-kubectl get cm powerscale-driver-v2.2.0 -n dell-csm-operator
+#!/bin/bash
 
-kubectl create cm driver-sidecars --from-file=sidecars/ -n dell-csm-operator
-k get cm driver-sidecars -n dell-csm-operator
+source ../common.bash
 
-kubectl create cm authorization-v1.2.0 --from-file=authorization/v1.2.0 -n dell-csm-operator
-k get cm authorization-v2.2.0 -n dell-csm-operator
+NS="dell-csm-operator"
+DRIVER="powerscale"
+CONFIGVERSION="v2.3.0"
+# All below variables not recommended for user modification
+DELETE_EXIST_CFGMAP=0
+TEMPDIR="/tmp/csm-oper-cfg-map/"
+PROG="${0}"
+CFGMAP_NAMES_FILE="cm_names.txt"
+SCRIPTDIR=`pwd`
+cfgmaps_created=""
+
+function cleanup() {
+  # Get rid of temp directory
+  rm -rf $TEMPDIR
+  cd $SCRIPTDIR
+}
+
+function cleanup_created_cfgmaps() {
+  if [ "$cfgmaps_created" != "" ]; then
+    kubectl delete configmap $cfgmaps_created -n $NS
+  fi
+}
+
+function read_configmap_names() {
+  # make sure there is no existing list
+  rm -rf $CFGMAP_NAMES_FILE
+  log section "Read configmap names from file"
+  cat $DRIVER-$CONFIGVERSION.yaml | while read line
+  do
+    echo $line | grep listOfConfigMapNames
+      if [ "$?" == "0" ]; then
+        echo "reading configmaps list"
+          while read line
+          do
+            echo $line | grep "-"
+            if [ "$?" == "0" ]; then
+              echo $line | awk ' {print $2}' >> $CFGMAP_NAMES_FILE
+            else
+              log step "Done reading configmap names"
+              break
+            fi
+          done
+        fi
+  done
+}
+
+function usage() {
+  echo
+  echo "Help for $PROG"
+  echo
+  echo "Usage: $PROG options..."
+  echo "Options:"
+  echo "  Optional"
+  echo "  --force-configmap-delete                 Force deletion of existing configmaps"
+  echo "  -h                                       Help"
+  echo
+
+  exit 0
+}
+
+while getopts ":h-:" optchar; do
+  case "${optchar}" in
+  -)
+    case "${OPTARG}" in
+    force-configmap-delete)
+      DELETE_EXIST_CFGMAP=1
+      ;;
+    *)
+      echo "Unknown option -${OPTARG}"
+      echo "For help, run $PROG -h"
+      exit 1
+      ;;
+    esac
+    ;;
+
+  h)
+    usage
+    ;;
+  *)
+    echo "Unknown option -${OPTARG}"
+    echo "For help, run $PROG -h"
+    exit 1
+    ;;
+  esac
+done
+
+# make sure kubectl is available
+kubectl --help >&/dev/null || {
+  log error "kubectl required for installation... exiting"
+  log Failed
+  exit 1
+}
+
+# Make temporary directory, move into it
+mkdir $TEMPDIR
+cd $TEMPDIR
+
+
+log section "Downloading configmap yamls"
+# Download config tgz -- you will be required to put your personal access token in here as long as the csm-operator-config repo is private
```

Comment:
you assume this script is run on a host with both git access and k8s access. 
please provide alternate , in case wget is done outside this script , accept path to script to tar tgx file and skip download